File: advanced/values-vs-messages/receiver.jl
# -*- coding: utf-8 -*-
using HELICS; const h = HELICS

function main()

    fedinfo = h.helicsCreateFederateInfo()
    h.helicsFederateInfoSetCoreType(fedinfo, h.HELICS_CORE_TYPE_ZMQ)
    h.helicsFederateInfoSetCoreInitString(fedinfo, "--loglevel=7")
    h.helicsFederateInfoSetTimeProperty(fedinfo, h.HELICS_PROPERTY_TIME_DELTA, 0.5)
    fed = h.helicsCreateCombinationFederate("receiver", fedinfo)
    endpoint = h.helicsFederateRegisterGlobalEndpoint(fed, "ep2", "")

    h.helicsFederateEnterExecutingMode(fed)
    currenttime = 0
    for t in 5:10
        while currenttime < t
            print("Requesting time")
            currenttime = h.helicsFederateRequestTime(fed, t)
            print("Finished requesting time")
        end
        while h.helicsEndpointHasMessage(endpoint)
            m = h.helicsEndpointGetMessageObject(endpoint)
            @show currenttime, h.helicsMessageGetString(m)
        end
    end

    h.helicsFederateFinalize(fed)
    h.helicsFederateFree(fed)
    h.helicsCloseLibrary()

end

main()

File: beginner/example-4/receiver.py
# -*- coding: utf-8 -*-
import os
import helics as h

fedinfo = h.helicsCreateFederateInfo()
h.helicsFederateInfoSetCoreType(fedinfo, h.helics_core_type_zmq)
h.helicsFederateInfoSetCoreInitString(fedinfo, "--loglevel=7")
h.helicsFederateInfoSetTimeProperty(fedinfo, h.helics_property_time_delta, 0.5)
fed = h.helicsCreateCombinationFederate("receiver", fedinfo,)
topicA = h.helicsFederateRegisterSubscription(fed, "topicA", "",)


h.helicsFederateEnterExecutingMode(fed)
currenttime = 0
for t in range(5, 10 + 1):
    while currenttime < t:
        currenttime = h.helicsFederateRequestTime(fed, t)
        a = h.helicsInputGetDouble(topicA)
        print(f"Received a = {a} at time = {currenttime}")

h.helicsFederateFinalize(fed)
h.helicsFederateFree(fed)
h.helicsCloseLibrary()
